src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -324,13 +339,27 @@ var ReactComponentTreeHook = {
     }
 
     var currentOwner = ReactCurrentOwner.current;
-    if (currentOwner && typeof currentOwner._debugID === 'number') {
-      var id = currentOwner && currentOwner._debugID;
-      info += ReactComponentTreeHook.getStackAddendumByID(id);
+    if (currentOwner) {
+      if (typeof currentOwner.tag === 'number') {
+        const fiber = ((currentOwner : any) : Fiber);
+        info += ReactComponentTreeHook.getStackAddendumByFiber(fiber);
+      } else if (typeof currentOwner._debugID === 'number') {
+        info += ReactComponentTreeHook.getStackAddendumByID(currentOwner._debugID);
+      }
     }
     return info;
   },
 
+  getStackAddendumByFiber(fiber : Fiber) : string {
+    var info = '';
+    var node = fiber;
+    do {
+      info += describeFiber(node);
+      node = node.return;
```

Comment:
My tentative plan is to figure out a way to get the tree hook working but I want to get the most important warnings (propTypes and keys) working by simpler means first. It gives me a better understanding of the problem to solve these separately (and maybe unify later) than try to unify right away.